src/timeline/TimelineModel.cpp

```diff
@@ -1934,11 +1934,279 @@ TimelineModel::formatPowerLevelEvent(const QString &id)
+                   (prevEvent->content.events_default > default_powerlevel)) {
+            resultingMessage.append(
+              tr("%1 has changed the room's events_default powerlevel from %2 to %3. New "
+                 "users can now send events that are not otherwise restricted.")
+                .arg(sender_name,
+                     QString::number(prevEvent->content.events_default),
+                     QString::number(event->content.events_default)));
+        } else {
+            resultingMessage.append(
+              tr("%1 has changed the room's events_default powerlevel from %2 to %3.")
+                .arg(sender_name,
+                     QString::number(prevEvent->content.events_default),
+                     QString::number(event->content.events_default)));
+        }
+    }
+
+    // Compare if a Powerlevel of a user changed
+    for (auto const &[mxid, powerlevel] : event->content.users) {
+        auto nameOfChangedUser = utils::replaceEmoji(displayName(QString::fromStdString(mxid)));
+        if (prevEvent->content.user_level(mxid) != powerlevel) {
+            if (powerlevel >= administrator_power_level) {
+                resultingMessage.append(tr("%1 has made %2 an administrator of this room.")
+                                          .arg(sender_name, nameOfChangedUser));
+            } else if (powerlevel >= moderator_power_level &&
+                       powerlevel > prevEvent->content.user_level(mxid)) {
+                resultingMessage.append(tr("%1 has made %2 a moderator of this room.")
+                                          .arg(sender_name, nameOfChangedUser));
+            } else if (powerlevel >= moderator_power_level &&
+                       powerlevel < prevEvent->content.user_level(mxid)) {
+                resultingMessage.append(tr("%1 has downgraded %2 to moderator of this room.")
+                                          .arg(sender_name, nameOfChangedUser));
+            } else {
+                resultingMessage.append(tr("%1 has changed the powerlevel of %2 from %3 to %4.")
+                                          .arg(sender_name,
+                                               nameOfChangedUser,
+                                               QString::number(prevEvent->content.user_level(mxid)),
+                                               QString::number(powerlevel)));
+            }
+        }
+    }
+
+    // Handle added/removed/changed event type
+    for (auto const &[event_type, powerlevel] : event->content.events) {
+        auto prev_not_present =
+          prevEvent->content.events.find(event_type) == prevEvent->content.events.end();
+
+        if (prev_not_present || prevEvent->content.events.at(event_type) != powerlevel) {
+            if (powerlevel >= administrator_power_level) {
+                resultingMessage.append(
+                  tr("%1 has made event type \"%2\" changeable only by admins.")
```

Comment:
changeable? Sendable? %1 allowed only admins to send %2?

Reply:
oops. :D